packages/plugin-vue/src/handleHotUpdate.ts

```diff
@@ -205,7 +205,7 @@ function hasScriptChanged(prev: SFCDescriptor, next: SFCDescriptor): boolean {
   // this is only available in vue@^3.2.23
   const prevImports = prevResolvedScript?.imports
   if (prevImports) {
-    return next.shouldForceReload(prevImports)
+    return !next.template || next.shouldForceReload(prevImports)
```

Comment:
For the test above, `!next.template` seems to be needed. I'm not sure whether this should be handled on vue side.

Reply:
I think we can have the condition here, and if it is handled later on vue we remove it. cc @sodatea